### ISSUE TITLE
For non-admin user "All Projects" shows only private projects

### ISSUE THREAD

**Issue by akshayl** [vmware/harbor#2040]
This user has 3 private projects and access to 6 public projects, but only the 3 private ones are shown when "All Projects" are selected:

![image](https://cloud.githubusercontent.com/assets/3515060/25114337/e3217670-23b2-11e7-9d71-75762eee3996.png)

**Comment by steven-zou** [comment]
@akshayl 

Just changed the label 'My Projects' to 'All Projects', no change on the data retrieving as the API does not support 'All' at the moment .
The API needs to be upgraded to support 'All Projects' functionality.

Involve @ywk253100 in.

**Comment by ywk253100** [comment]
The API called by "All Projects" returns all projects which the user is a member of, include both private and public projects. 
We should confirm whether the tab is "All Projects" which is meaning all my private projects and all public projects or is "All My Projects" which is shown correctly now first.

**Comment by lweitzman** [comment]
i thought that "personal projects" (was called "my projects") are projects that i "own". they could be public projects but ones that i own. is this correct? i.e., "personal" is not the same as private.

**Comment by akshayl** [comment]
"All Projects" should show all the projects that the user is a member of whether or not they are public or private.

**Comment by ywk253100** [comment]
@akshayl I think it's the current behaviour for "All Projects" now. Is it right @steven-zou ?
@akshayl Isn't it what you see on your Harbor instance? Could you provide the environment information so that we can check on it ?

**Comment by lweitzman** [comment]
in the latest build at https://10.117.4.151/harbor/projects "all projects" is showing all 19 projects and public projects is showing 6 projects. both are correct based on the metrics in the upper right. @akshayl should we be also allowing the user to filter "Personal Projects" as well as "Public Projects" and "All Projects"?

**Comment by steven-zou** [comment]
@akshayl @lweitzman 

  Reach any agreement?

**Comment by akshayl** [comment]
There seems to be a lot of confusion around this. I'll work with Louis offline to clarify things.

**Comment by lweitzman** [comment]
after speaking to Akshay, the resolution is the use Private - Public - Total in the metrics of the projects page. the dropdown will have the options "All Projects", "Private Projects", "Public Projects"

![screen shot 2017-04-24 at 3 38 19 pm](https://cloud.githubusercontent.com/assets/626032/25355348/52e132d8-2904-11e7-8706-b7b8e7f59749.png)

**Comment by reasonerjt** [comment]
@lweitzman @akshayl Sorry but I feel the term "private" misleading, esp. when it comes to repositories, it sounds like they belong to the user himself, but in fact it's shared within project.

Is there reason not to use "My projects"/"Public projects"?  Is it necessary to highlight the projects that are not public?

**Comment by akshayl** [comment]
I see Harbor projects as similar to GitHub repos. In GitHub, repos can be public (open to everyone) or private (only certain people have access to them). Similarly, in Harbor we have public projects (any user can access them) and private project (restricted to a certain set of users).

**Comment by akshayl** [comment]
In the mockup the total should be total of private + public. So the numbers for total should be 5 and 10 respectively.

**Comment by steven-zou** [comment]
@akshayl @lweitzman @reasonerjt 

  I think confusion existing between project 'access level' property and project 'owner' in current implementation.

   Should clarify the different usage case.

  For 'private' case, only show the projects with condition matched 'access_level === private' ;
  For 'public' case, only show the projects with condition matched 'access_level === public' ;
 For 'all' case, show both private and public projects.

 We can add one more pre-defined filter 'My Projects' in the dropdown menu list, display all the projects whose owner is me (or a 'projectAdmin' role is assigned to me in the project etc.).

**Comment by stuclem** [comment]
@akshayl @reasonerjt @steven-zou do we need to add this to the release notes for 1.1?

**Comment by steven-zou** [comment]
After going through the comments if the issue, I found there might be existing some misunderstandings.
 
  We have agreed to change the project property ‘public’ to ‘Access level’ which has two unique values ‘public’ and ‘private’. ‘Access level’ defines the ‘publicity’ of the project. ‘public’ means everyone can see it. ‘private’ limits the access scope to the specified users (users should be members of the project).
 
  The default ‘projects’ view shows all the projects which the current logged user is member of. Either is ‘private’ access level or ‘public’ access level. As the filter dropdown list is lay close to the project list view, the filtering scope should be limited to the list. And then, by my understanding, the filers should be:
 
   ‘All projects’: Show all the projects of mine. (That means I’m member of those projects)
   ‘Private projects’: Only show my projects whose access levels are ‘private’.
   ‘Public projects’:  Only show my projects whose access levels are ‘public’.
 
  In the old version of Harbor portal, there are two different tabs in project view, one is ‘My Projects’ like current view and another is ‘Public Projects’. The list in tab ‘Public Projects’ shows all the projects whose ‘access level’ property is ‘public’ in the system. The current logged user might be member of some of those projects, but not ALL, sometimes might be none.
 
  We might miss above case in the new UI.
 
   Additionally, the above issue is also influenced the project metrics. The ‘TOTAL” number is only for administrators and means the total number of the projects existing in the whole system. The ‘PUBLIC” number means the number of all the projects whose access level is ‘public’ in the whole system. The ‘MY’ number means the number of all my (member of project) projects. And then ‘MY’+’PUBLIC’ <> ‘TOTAL’.
 
  In summary, we might need to consider adjusting the UI design to reflect the actual meanings. Obviously, I think it seems a little hard to deliver the fix of this issue in rel 1.1.1.

**Comment by steven-zou** [comment]
From Brad:
===========

For 1.1.1, let’s just fix the label problem and not adjust the logic. So in the dashboard section at the top, on the projects row, let’s change “My” to “My Projects” and on the repositories row, let’s change “My” to “My Repositories.” For the column header that currently says “Public” let’s change it to “Access Level.” Any objections to making this set of cosmetic change in 1.1.1? For 1.2, we have more time to discuss and adjust the logic if necessary.

**Comment by reasonerjt** [comment]
For 1.1.1 only the cosmetic change will be made.

**Comment by ali5ter** [comment]
at the risk of putting my oar in ;)

### rational
I would hypothesis that the majority of developer expectation and experience comes from their use of github. They are use to filtering on All, Private and Public. Developers are used to a Private repo being restricted access and public having no limitations of access.

Using the github model seems reasonable. If you want to compare with DockerHub, you'll see they don't even filter on access in the UI.

### recommendation
* Having ability to see All, Private and Public seems reasonable.
* The column titled 'Public' should read 'Access' as in access level.
* The label in the top left of the Projects screen that reads 'my', should read 'private'
* I'd also remove the total from the top right for all users. I don't see a reason to show that when you're already showing numbers for the access types.

**Comment by steven-zou** [comment]
@ali5ter 
  Thanks a lot for the valuable comments. We'll deliver a proper reasonable solution to clarify the confusions of project filtering in next release.

**Comment by steven-zou** [comment]
A proposal:
  To provide more flexible and powerful project filtering capability,  I suggest to add two select boxes. One is for the topic of whether the current user is member/owner of the project. Another one is for the access level, like current one. 

Select box 1:
   All //No matter what roles I am
   I'm member
   I'm creator

Select box 2:
   All
   Public
   Private

The two filters will applied to the filtering together. Then we'll get several filtering scenario:

box1==All && box2==All : All projects I can See.
box1==All && box2==Public : All public projects.
box1==All && box2==Private : All private projects I can See.
box1==I'm member && box2==All : All projects of which I'm a member.
box1==I'm member && box2==Public : All public projects of which I'm a member.
box1==I'm member && box2==Private : All private projects I can See.
box1==I'm creator && box2==All : All projects of which I'm a creator.
box1==I'm creator && box2==Public : All public projects of which I'm a creator.
box1==I'm creator && box2==Private : All private projects of which I'm a creator.

@akshayl @lweitzman @tonycgil

**Comment by lweitzman** [comment]
@steven-zou i think that is too many widgets and not sure the workflows require 2 filters (even though it is more flexible). in my opinion, we could just have  select box 2: all, public, private and that should be sufficient.

**Comment by steven-zou** [comment]
Changes in PR #2516. Close this issue.
